networkx/algorithms/approximation/connectivity.py

```diff
@@ -244,6 +244,18 @@ def all_pairs_node_connectivity(G, nbunch=None, cutoff=None):
     K : dictionary
         Dictionary, keyed by source and target, of pairwise node connectivity
 
+    Examples
+    --------
+    >>> # A 3 node cycle with one extra node attached has connectivity 2 between all
+    >>> # nodes in the cycle and connectivity 1 between the extra node and the rest
+    >>> G = nx.cycle_graph(3)
+    >>> G.add_edge(2, 3)
+    >>> nx.all_pairs_node_connectivity(G)
+    {0: {1: 2, 2: 2, 3: 1},
+    1: {0: 2, 2: 2, 3: 1},
+    2: {0: 2, 1: 2, 3: 1},
+    3: {0: 1, 1: 1, 2: 1}}
```

Comment:
The tests are failing on the output here because `doctest` requires that [the output characters be exactly what is given by the Python interpreter](https://docs.python.org/3/library/doctest.html#how-it-works). This is a common problem that catches basically everyone who is not used to doctest's quirks.

There are a couple options here - you could import `pprint` and `pprint` the result, which is what gives the nicely-formatted output that you saw in your terminal (I'm guessing IPython or a Jupyter notebook):

```suggestion
    >>> import pprint  # for nice dictionary formatting
    >>> pprint.pprint(nx.all_pairs_node_connectivity(G))
    {0: {1: 2, 2: 2, 3: 1},
    1: {0: 2, 2: 2, 3: 1},
    2: {0: 2, 1: 2, 3: 1},
    3: {0: 1, 1: 1, 2: 1}}
```

... or, if you think importing a module just for printing distracts from the example, you can handle things manually:

```python
    >>> for n, conn in nx.all_pairs_node_connectivity(G).items():
    ...     print(f"{n} : {conn}")
    0 : {1: 2, 2: 2, 3: 1}
    1 : {0: 2, 2: 2, 3: 1}
    2 : {0: 2, 1: 2, 3: 1}
    3 : {0: 1, 1: 1, 2: 1}
```

Reply:
I was wondering why that happened! Thanks @rossbar!